contracts/refs/CoreRefV2.sol

```diff
@@ -0,0 +1,241 @@
+    modifier hasAnyOfTwoRoles(bytes32 role1, bytes32 role2) {
+        require(
+            _core.hasRole(role1, msg.sender) ||
+                _core.hasRole(role2, msg.sender),
+            "UNAUTHORIZED"
+        );
+        _;
+    }
+
+    modifier hasAnyOfThreeRoles(
+        bytes32 role1,
+        bytes32 role2,
+        bytes32 role3
+    ) {
+        require(
+            _core.hasRole(role1, msg.sender) ||
+                _core.hasRole(role2, msg.sender) ||
+                _core.hasRole(role3, msg.sender),
+            "UNAUTHORIZED"
+        );
+        _;
+    }
+
+    modifier hasAnyOfFourRoles(
+        bytes32 role1,
+        bytes32 role2,
+        bytes32 role3,
+        bytes32 role4
+    ) {
+        require(
+            _core.hasRole(role1, msg.sender) ||
+                _core.hasRole(role2, msg.sender) ||
+                _core.hasRole(role3, msg.sender) ||
+                _core.hasRole(role4, msg.sender),
+            "UNAUTHORIZED"
+        );
+        _;
+    }
+
+    modifier hasAnyOfFiveRoles(
+        bytes32 role1,
+        bytes32 role2,
+        bytes32 role3,
+        bytes32 role4,
+        bytes32 role5
+    ) {
+        require(
+            _core.hasRole(role1, msg.sender) ||
+                _core.hasRole(role2, msg.sender) ||
+                _core.hasRole(role3, msg.sender) ||
+                _core.hasRole(role4, msg.sender) ||
+                _core.hasRole(role5, msg.sender),
+            "UNAUTHORIZED"
+        );
+        _;
+    }
+
+    /// @notice set pausable methods to paused
+    function pause() public override onlyGuardianOrGovernor {
+        _pause();
+    }
+
+    /// @notice set pausable methods to unpaused
+    function unpause() public override onlyGuardianOrGovernor {
+        _unpause();
+    }
+
+    /// @notice address of the Core contract referenced
+    /// @return ICore implementation address
+    function core() public view override returns (ICoreV2) {
+        return _core;
+    }
+
+    /// @notice address of the Volt contract referenced by Core
+    /// @return IVolt implementation address
+    function volt() public view override returns (IVolt) {
+        return _core.volt();
+    }
+
+    /// @notice address of the Vcon contract referenced by Core
+    /// @return IERC20 implementation address
+    function vcon() public view override returns (IERC20) {
+        return _core.vcon();
+    }
+
+    /// @notice volt balance of contract
+    /// @return volt amount held
+    function voltBalance() public view override returns (uint256) {
+        return volt().balanceOf(address(this));
+    }
+
+    /// @notice vcon balance of contract
+    /// @return vcon amount held
+    function vconBalance() public view override returns (uint256) {
+        return vcon().balanceOf(address(this));
+    }
+
+    /// @notice set new reference to core
+    /// only callable by governor
+    /// @param newCore to reference
+    function setCore(address newCore) external onlyGovernor {
```

Comment:
I would make it very clear with asterisks or comments or similar the magnitude of calling this. It would potentially brick the protocol if done incorrectly, looks surprisingly innocuous

Reply:
warning added